internal/ingress/types.go

```diff
@@ -85,6 +85,30 @@ type Backend struct {
 	UpstreamHashBy string `json:"upstream-hash-by,omitempty"`
 	// LB algorithm configuration per ingress
 	LoadBalancing string `json:"load-balance,omitempty"`
+	// Denotes if a backend has no server. The backend instead shares a server with another backend and acts as an
+	// alternative backend.
+	// This can be used to share multiple upstreams in the sam nginx server block.
+	NoServer bool `json:"hasServer"`
+	// Policies to describe the characteristics of an alternative backend.
+	// +optional
+	TrafficShapingPolicy TrafficShapingPolicy `json:"trafficShapingPolicy,omitempty"`
+	// Contains a list of backends without servers that are associated with this backend.
+	// +optional
+	AlternativeBackends []*Backend `json:"alternativeBackends,omitempty"`
```

Comment:
There should be a way to avoid sending the whole backend structs here, they are already being posted as normal backends to Lua land. Only their names should be needed here.

Reply:
I can make that work. Since every backend has a balancer in lua land, keeping just their names should be straightforward to change.

Reply:
The tricky part is that you won't have access to traffic shaping policies then. I guess that can be part of the respective balancer instance. Then in Lua code instead of `alternative_backend` you can work with `alternative_balancer`.